File: src/lib/code/mcs-216/section-1/session-1/1/1.py
def gcd_with_stats(a, b):
    """
    Computes GCD and tracks modulo and assignment operations.
    """
    mod_count = 0
    assign_count = 0

    # Absolute values for mathematical consistency
    a, b = abs(a), abs(b)
    assign_count += 2

    while b:
        # Modulo operation
        remainder = a % b
        mod_count += 1

        # Assignment operations (a = b, b = remainder)
        a = b
        b = remainder
        assign_count += 2

    return a, mod_count, assign_count

# Input values
val_a, val_b = 15265, 15
result, mods, assigns = gcd_with_stats(val_a, val_b)

print(f"GCD({val_a}, {val_b}) = {result}")
print(f"Modulo Operations: {mods}")
print(f"Assignment Operations: {assigns}")

File: src/lib/code/mcs-216/section-1/session-1/1/1.rs
fn gcd_with_metrics(mut a: i64, mut b: i64) -> (i64, u32, u32) {
    let mut mod_ops = 0;
    let mut assign_ops = 0;

    // Initial assignments to absolute values
    a = a.abs();
    b = b.abs();
    assign_ops += 2;

    while b != 0 {
        let remainder = a % b;
        mod_ops += 1;

        a = b;
        b = remainder;
        assign_ops += 2;
    }

    (a, mod_ops, assign_ops)
}

fn main() {
    let (result, mods, assigns) = gcd_with_metrics(15265, 15);
    println!("GCD: {}", result);
    println!("Modulo Ops: {}, Assignment Ops: {}", mods, assigns);
}
